python/lsst/atmospec/spectraction.py

```diff
@@ -133,22 +138,38 @@ def dumpParameters():
             if not item.startswith("__"):
                 print(item, getattr(parameters, item))
 
-    def spectractorImageFromLsstExposure(self, exp, *, target_label='', disperser_label='', filter_label=''):
+    def debugPrintTargetCentroidValue(self, image):
+        x, y = image.target_guess
+        self.log.debug(f"Image shape = {image.data.shape}")
+        self.log.debug(f"x, y = {x}, {y}")
+        x = int(np.round(x))
+        y = int(np.round(y))
+        self.log.debug(f"Value at {x}, {y} = {image.data[y, x]}")
+
+    def spectractorImageFromLsstExposure(self, exp, xpos, ypos, *, target_label='',
+                                         disperser_label='', filter_label=''):
         """Construct a Spectractor Image object from LSST objects.
 
         Internally we try to use functions that calculate things and return
         them and set the values using the return rather than modifying the
         object in place where possible. Where this is not possible the methods
         are labeled _setSomething().
         """
+        # TODO: passing exact centroids seems to be causing a serious
+        # and non-obvious problem!
+        # this needs fixing for several reasons, mostly because if we have a
+        # known good centroid then we want to skip the refitting entirely
```

Comment:
Also docstrings.  As part of the review below, I'm thinking through the results of this call, and had to track back to here and then to the header to realize that the `image` is a `spectractor.extractor.images.Image` and not one of the many other types we have with `Image` in their name.

Reply:
Yup, sorry about that. Will do a doc overhaul before merging.